libpius/mailer.py

```diff
@@ -252,36 +254,43 @@ def _send_mail(self, to, msg):
       msg['To'] = to
     msg['Date'] = formatdate(localtime=True)
 
-    try:
-      smtp = smtplib.SMTP(self.host, self.port)
-      if self.tls:
-        # NOTE WELL: SECURITY IMPORTANT NOTE!
-        # In python 2.6 if you attempt to starttls() and the server doesn't
-        # understand an exception is raised. However before that, it just
-        # carried on # and one could attempt to auth over a plain-text session.
-        # This is BAD!
-        #
-        # So, in order be secure on older pythons we ehlo() and then check the
-        # response before attempting startls.
-        smtp.ehlo()
-        if not smtp.has_extn('STARTTLS'):
-          # Emulate 2.6 behavior
-          raise smtplib.SMTPException('Server does not support STARTTLS')
-        smtp.starttls()
-        # must re-ehlo after STARTTLS
-        smtp.ehlo()
-        # Don't want to send auth information unless we're TLS'd
-        if self.user:
-          smtp.login(self.user, self.password)
-      if self.address_override:
-        env_to = self.address_override
-      else:
-        # BCC the user...
-        env_to = [msg['To'], self.mail]
+    if self.local_mail_dir:
+      if not os.path.isdir(self.local_mail_dir):
+        os.mkdir(self.local_mail_dir)
+      email = open(os.path.join(self.local_mail_dir, msg['To']), 'w')
+      email.write(str(msg))
+      email.close()
+    else:
+      try:
+        smtp = smtplib.SMTP(self.host, self.port)
+        if self.tls:
+          # NOTE WELL: SECURITY IMPORTANT NOTE!
+          # In python 2.6 if you attempt to starttls() and the server doesn't
+          # understand an exception is raised. However before that, it just
+          # carried on # and one could attempt to auth over a plain-text session.
+          # This is BAD!
+          #
+          # So, in order be secure on older pythons we ehlo() and then check the
+          # response before attempting startls.
+          smtp.ehlo()
+          if not smtp.has_extn('STARTTLS'):
+            # Emulate 2.6 behavior
+            raise smtplib.SMTPException('Server does not support STARTTLS')
+          smtp.starttls()
+          # must re-ehlo after STARTTLS
+          smtp.ehlo()
+          # Don't want to send auth information unless we're TLS'd
+          if self.user:
```

Comment:
The more I think about this approach the more I like it.

However... how do we handle the email_override and the BCC-the-user bit here?

Reply:
But you aren't actually writing it to the file... env_to isn't in the object you write out.

Reply:
Yeah, keeping py2 compat for now. BUt also as mentioned I think we should only add that Bcc header to the message when we're saving it as a proxy for a proper envelope header.